### PR TITLE
Bug/burn doesnt delete resources

### LINKED ISSUE #80: burnNft leaves resources
burnNft successfully destroys the NFT storage for the NFT, but Resources is stored separately (in Resources storage).  burn operation doesn't destroy this storage, which it should.

repro:
create collection
create nft
add resource to nft
burn nft
check resources storage (it will still be there)

### PR DESCRIPTION
**Pull request by bmacer**
fixes #80

### CODE REVIEW

File: pallets/rmrk-core/src/functions.rs [hunk 1]
```diff
@@ -233,6 +233,9 @@ where
 	) -> sp_std::result::Result<(CollectionId, NftId), DispatchError> {
 		ensure!(max_recursions > 0, Error::<T>::TooManyRecursions);
 		Nfts::<T>::remove(collection_id, nft_id);
+		for resource in Resources::<T>::iter_prefix((collection_id, nft_id)) {
+			Resources::<T>::remove((collection_id, nft_id, resource.1.id));
+		}
```
Comment (h4x3rotab) [inline pallets/rmrk-core/src/functions.rs:238]: Mutating the storage map while iterating over it. Not sure if this is problematic. I asked a question in substarte stackexchange:

https://substrate.stackexchange.com/questions/573/is-it-safe-to-delete-elements-when-iterating-through-a-storage-map-in-substrate

Comment (h4x3rotab) [reply pallets/rmrk-core/src/functions.rs:238]: According to the answer, we should use `drain_prefix`

Comment (ilionic) [reply pallets/rmrk-core/src/functions.rs:238]: Good to know, also found this usage pattern with `drain_prefix` in uniques pallet

Comment (h4x3rotab) [reply pallets/rmrk-core/src/functions.rs]: Oh, while using `drain_prefix` we don't need to call `Resource::<T>::remove` any more

Comment (ilionic) [reply pallets/rmrk-core/src/functions.rs]: Got it, something like that then?
```
for _ in Resources::<T>::drain_prefix((collection_id, nft_id)) {}		
```

Comment (h4x3rotab) [reply pallets/rmrk-core/src/functions.rs]: We can safely ignore its return value like this:

```
Resources::<T>::drain_prefix((collection_id, nft_id));
```

**Pull request merged**